tests/install/steps/test_expansion.py

```diff
@@ -0,0 +1,135 @@
+import json
+import pathlib
+import string
+
+import pytest
+from pytest_bdd import scenario, then, parsers, when
+import testinfra
+import kubernetes as k8s
+import yaml
+
+from tests import kube_utils
+from tests import utils
+
+# Scenarios
+@pytest.mark.skip(reason='waiting for #1038')
+@scenario('../features/expansion.feature',
+          'Add one node to the cluster',
+          strict_gherkin=False)
+def test_cluster_expansion(host):
+    pass
+
+# When {{{
+
+@when(parsers.parse('we declare a new "{node_type}" node on host "{hostname}"'))
+def declare_node(request, version, k8s_client, node_type, hostname):
+    """Declare the given node in Kubernetes."""
+    ssh_config = request.config.getoption('--ssh-config')
+    node_ip = get_node_ip(hostname, ssh_config)
+    node_manifest = get_node_manifest(node_type, version, node_ip)
+    k8s_client.create_node(body=node_from_manifest(node_manifest))
+
+
+@when(parsers.parse('we deploy the node "{name}"'))
+def deploy_node(version, k8s_client, name):
+    accept_ssh_key = [
+        'salt-ssh', '-i', name, 'test.ping', '--roster=kubernetes'
+    ]
+    pillar = {'bootstrap_id': 'bootstrap', 'node_name': name}
+    deploy = [
+        'salt-run', 'state.orch', 'metalk8s.orchestrate.deploy_new_node',
+        'saltenv=metalk8s-{}'.format(version),
+        'pillar={}'.format(json.dumps(pillar))
+    ]
+    run_salt_command(k8s_client, accept_ssh_key)
+    run_salt_command(k8s_client, deploy)
+
+
+# }}}
+# Then {{{
+
+@then(parsers.parse('node "{hostname}" is registered in Kubernetes'))
+def check_node_is_registered(k8s_client, hostname):
+    """Check if the given node is registered in Kubernetes."""
+    selector = 'metadata.name=={}'.format(hostname)
+    nodes = k8s_client.list_node(field_selector=selector).items
+    assert len(nodes) <= 1, 'too many nodes with hostname `{}`'.format(hostname)
+    assert len(nodes) == 1, '`{}` not registered in Kubernetes'.format(hostname)
+
+@then(parsers.parse(
```

Comment:
That's a copy paste from `post` features and I'm not happy with that.

@gdemonet any way to reuse steps between test suites?

Reply:
Yep, just define them in a `conftest.py` to make sure they are available. `given`/`when`/`then` are actually fixtures, IIUC.

See https://pytest-bdd.readthedocs.io/en/stable/#reusing-steps

Reply:
Yeah of course, I even already did it for `@given("the Kubernetes API is available")` >_<